### PR TITLE
Update setup docker

### PR DESCRIPTION
**Pull request by tagoylo**
This PR includes changes that updates setting up the docker container and how nebula, libiio, and telemetry are updated if needed.

1. The default docker image used is now set to test-harness-ci:latest. This image already has pre-installed nebula, libiio, and telemetry, all are based from master branch.
2. Reinstalling the libraries ( nebula, libiio, and telemetry ) upon setup is also possible by still using the set_env method to set the repo link and repo branch. A function to check if there is difference of the newly set and default is placed to see if there is a need to update the libraries: _check_update_container_lib_
3. If needed to update the libraries ( nebula, libiio, and telemetry ) upon setup base on master, it is possible to use the  _update_container_lib_ and set it to true. This will force to update the libraries base on master branch.
4. A small change was also implemented in updating the libraries on the agent itself, it is now default to not reinstall the requirements.txt but may set it to do so by setting _update_lib_requirements_ to true.

### CODE REVIEW

File: src/sdg/Gauntlet.groovy [hunk 1]
```diff
@@ -929,18 +930,15 @@ private def run_agents() {
...
                             sh 'cp /default/pydistutils.cfg /root/.pydistutils.cfg || true'
                             sh 'mkdir -p /root/.config/pip && cp /default/pip.conf /root/.config/pip/pip.conf || true'
                             sh 'cp /default/pyadi_test.yaml /etc/default/pyadi_test.yaml || true'
-                            sh 'cp -r /app/* "${PWD}"/'
-                            setup_locale()
-                            setup_libserialport()
-                            setupAgent(['libiio','nebula','telemetry'], true, docker_status);
+                            def deps = check_update_container_lib()
+                            if (deps.size()>0){
+                                setupAgent(deps, true, false)
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:940]: MIght need to also parametrize the 3rd argument (update_requirements) ?

File: src/sdg/Gauntlet.groovy [hunk 2]
```diff
@@ -1761,36 +1738,42 @@ private def setup_libserialport() {
     }
 }
 
-private def setupAgent(deps, skip_cleanup = false, docker_status) {
+private def check_update_container_lib(update_container_lib=false) {
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:1741]: cool implementation 😎

File: src/sdg/Gauntlet.groovy [hunk 3]
```diff
@@ -1761,36 +1738,42 @@ private def setup_libserialport() {
...
+    def deps = []
+    def default_branch = 'master'
+    def default_repos = ['https://github.com/sdgtt/nebula.git', 'https://github.com/analogdevicesinc/libiio.git', 'https://github.com/sdgtt/telemetry.git']
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:1744]: Is groovy list order guaranteed? (not sure but documentation says so) but to be safe, might also be good to have the same implementation as line 1747 in defining these lists

**Comment by tagoylo** [comment]
Updated the implementation of updating the libraries inside the container:
1. set update_container_lib to true to update the libraries.
2. set update_lib_requirements to true to install again requirements.txt.
3. check_update_container_lib will only check for branches, if the branch is not master it will automatically update the library inside the container.
4. In case the user wants to always check or get the newest commits of the libraries, set update_container_lib to true. Since the docker image is only built once a week.

@jpineda3 @kimpaller

**Pull request merged**